runtime/autoload/man.vim

```diff
@@ -79,43 +79,61 @@ function! man#read_page(ref) abort
   call s:put_page(page)
 endfunction
 
-" Handler for s:system() function.
-function! s:system_handler(jobid, data, event) dict abort
-  if a:event is# 'stdout' || a:event is# 'stderr'
-    let self[a:event] .= join(a:data, "\n")
-  else
-    let self.exit_code = a:data
+" Run man and timeout after 30 seconds.
```

Comment:
Let's avoid any refactor here for now and get all the function changes in first.

Reply:
Oh man, I thought it would be easy to "pick the commits" without having to "pick the whole PR". I guess I'll better create a new branch for each commit? I'm not sure :disappointed: